job-service/server/src/main/java/org/verapdf/webapp/jobservice/server/service/JobService.java

```diff
@@ -51,6 +53,26 @@ public JobDTO getJobById(UUID jobId) throws NotFoundException {
 		return jobMapper.createDTOFromEntity(job);
 	}
 
+	@Transactional
+	public JobDTO updateJob(UUID jobId, JobDTO jobDTO) throws NotFoundException, BadRequestException {
+		Job job = findJobById(jobId);
+		if (job.getStatus() != JobStatus.CREATED) {
+			LOGGER.info("Cannot update job:" + job.getId() + ". Job is being processed or have been processed already.");
+			throw new BadRequestException("Cannot update job with specified id: " +
```

Comment:
It is better to throw something like ConflictException and handle it with 409 status (create new common exception and add handling in the main handler)